clientlibrary/worker/worker.go

```diff
@@ -269,37 +266,72 @@ func (w *Worker) eventLoop() {
 					continue
 				}
 
-				err = w.checkpointer.GetLease(shard, w.workerID)
-				if err != nil {
-					// cannot get lease on the shard
-					if err.Error() != chk.ErrLeaseNotAquired {
-						log.Error(err)
-					}
+				acquired := w.acquire(shard)
+				if !acquired {
 					continue
 				}
 
-				// log metrics on got lease
-				w.mService.LeaseGained(shard.ID)
-
-				log.Infof("Start Shard Consumer for shard: %v", shard.ID)
-				sc := w.newShardConsumer(shard)
-				go sc.getRecords(shard)
-				w.waitGroup.Add(1)
-				// exit from for loop and not to grab more shard for now.
-				break
+				leasesStolen++
+				if leasesStolen >= w.kclConfig.MaxLeasesToStealAtOneTime {
```

Comment:
honor the max leases to steal configuration

Reply:
do we need any locking here?